### ISSUE TITLE
Error with function plot_wrapper_DE_results()

### ISSUE THREAD

**Issue by fpichon** [Ylefol/TimeSeriesAnalysis#7]
Hi,
Thanks for your tool, it's very useful!
But recently, I have an error with the function plot_wrapper_DE_results(), with both 'conditional' and 'temporal' data:

```
Error in grid.Call.graphics(C_downvppath, name$path, name$name, strict): Viewport 'log10(counts)_heatmap_body_1_1' was not found
Traceback:

1. plot_wrapper_DE_results(object = TS_object, DE_type = "temporal", 
 .     genes_of_interest = genes_of_interest, results_folder = path2)
2. custom_heatmap_wrapper(object, DE_type = DE_type, log_transform = TRUE, 
 .     plot_file_name = paste0(main_path, "custom_heat_", DE_type), 
 .     adjust_missing_temp_samples, do_SVGs = do_SVGs)
3. plot_custom_DE_heatmap(my_heat_mat, my_region_split, my_group_split, 
 .     my_l2fc_vect, log_transform = log_transform, legend_value = legend_val, 
 .     plot_file_name = plot_file_name, do_SVG = do_SVGs)
4. draw(Heatmap(heat_mat, name = count_legend, cluster_rows = clust_rows, 
 .     cluster_columns = clust_cols, show_column_names = FALSE, 
 .     show_row_names = FALSE, row_names_side = "left", row_split = row_splits, 
 .     column_split = col_split, border = TRUE, na_col = "gray", 
 .     column_title = NULL, row_title = NULL, top_annotation = top_annot, 
 .     left_annotation = left_annot, bottom_annotation = bottom_histo, 
 .     cluster_column_slices = FALSE, cluster_row_slices = FALSE), 
 .     annotation_legend_list = lgd, annotation_legend_side = "bottom")
5. draw(Heatmap(heat_mat, name = count_legend, cluster_rows = clust_rows, 
 .     cluster_columns = clust_cols, show_column_names = FALSE, 
 .     show_row_names = FALSE, row_names_side = "left", row_split = row_splits, 
 .     column_split = col_split, border = TRUE, na_col = "gray", 
 .     column_title = NULL, row_title = NULL, top_annotation = top_annot, 
 .     left_annotation = left_annot, bottom_annotation = bottom_histo, 
 .     cluster_column_slices = FALSE, cluster_row_slices = FALSE), 
 .     annotation_legend_list = lgd, annotation_legend_side = "bottom")
6. .local(object, ...)
7. draw(ht_list, ...)
8. draw(ht_list, ...)
9. .local(object, ...)
10. grid.draw(p)
11. grid.draw.gTree(p)
12. recordGraphics(drawGTree(x), list(x = x), getNamespace("grid"))
13. drawGTree(x)
14. grid.draw(x$children[[i]], recording = FALSE)
15. grid.draw.grob(x$children[[i]], recording = FALSE)
16. recordGraphics(drawGrob(x), list(x = x), getNamespace("grid"))
17. drawGrob(x)
18. preDraw(x)
19. preDraw.grob(x)
20. pushvpgp(x)
21. pushgrobvp(x$vp)
22. pushgrobvp.vpPath(x$vp)
23. downViewport(vp, strict = TRUE, recording = FALSE)
24. downViewport.vpPath(vp, strict = TRUE, recording = FALSE)
25. grid.Call.graphics(C_downvppath, name$path, name$name, strict)
```

I tried to use different versions of ComplexHeatmap but it didn't resolve the issue..

Here is my setup:

```
R version 4.3.2 (2023-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=French_France.utf8  LC_CTYPE=French_France.utf8   
[3] LC_MONETARY=French_France.utf8 LC_NUMERIC=C                  
[5] LC_TIME=French_France.utf8    

time zone: Europe/Paris
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] org.Sc.sgd.db_3.18.0        AnnotationDbi_1.64.1       
 [3] pheatmap_1.0.12             PCAtools_2.14.0            
 [5] ggrepel_0.9.5               ggplot2_3.4.4              
 [7] TimeSeriesAnalysis_01.01.01 BiocFileCache_2.10.1       
 [9] dbplyr_2.4.0                stringr_1.5.1              
[11] tidyr_1.3.1                 DESeq2_1.42.0              
[13] SummarizedExperiment_1.32.0 Biobase_2.62.0             
[15] MatrixGenerics_1.14.0       matrixStats_1.2.0          
[17] GenomicRanges_1.54.1        GenomeInfoDb_1.38.5        
[19] IRanges_2.36.0              S4Vectors_0.40.2           
[21] BiocGenerics_0.48.1        

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3        rstudioapi_0.15.0        
  [3] jsonlite_1.8.8            shape_1.4.6              
  [5] tximport_1.30.0           magrittr_2.0.3           
  [7] farver_2.1.1              GlobalOptions_0.1.2      
  [9] fs_1.6.3                  zlibbioc_1.48.0          
 [11] vctrs_0.6.5               DelayedMatrixStats_1.24.0
 [13] memoise_2.0.1             RCurl_1.98-1.14          
 [15] base64enc_0.1-3           htmltools_0.5.7          
 [17] S4Arrays_1.2.0            dynamicTreeCut_1.63-1    
 [19] curl_5.2.0                tictoc_1.2               
 [21] SparseArray_1.2.3         htmlwidgets_1.6.4        
 [23] plyr_1.8.9                plotly_4.10.4            
 [25] cachem_1.0.8              uuid_1.2-0               
 [27] tximportData_1.30.0       lifecycle_1.0.4          
 [29] iterators_1.0.14          pkgconfig_2.0.3          
 [31] rsvd_1.0.5                Matrix_1.6-5             
 [33] R6_2.5.1                  fastmap_1.1.1            
 [35] GenomeInfoDbData_1.2.11   clue_0.3-65              
 [37] digest_0.6.34             colorspace_2.1-0         
 [39] dqrng_0.3.2               irlba_2.3.5.1            
 [41] RSQLite_2.3.5             beachmat_2.18.0          
 [43] org.Hs.eg.db_3.18.0       labeling_0.4.3           
 [45] filelock_1.0.3            org.Mm.eg.db_3.18.0      
 [47] fansi_1.0.6               httr_1.4.7               
 [49] abind_1.4-5               compiler_4.3.2           
 [51] bit64_4.0.5               withr_3.0.0              
 [53] doParallel_1.0.17         BiocParallel_1.36.0      
 [55] DBI_1.2.1                 DelayedArray_0.28.0      
 [57] rjson_0.2.21              tools_4.3.2              
 [59] glue_1.7.0                GOSemSim_2.28.1          
 [61] pbdZMQ_0.3-11             cluster_2.1.6            
 [63] reshape2_1.4.4            generics_0.1.3           
 [65] gtable_0.3.4              tzdb_0.4.0               
 [67] data.table_1.15.0         hms_1.1.3                
 [69] ScaledMatrix_1.10.0       BiocSingular_1.18.0      
 [71] utf8_1.2.4                XVector_0.42.0           
 [73] foreach_1.5.2             pillar_1.9.0             
 [75] yulab.utils_0.1.4         IRdisplay_1.1            
 [77] limma_3.58.1              circlize_0.4.15          
 [79] dplyr_1.1.4               lattice_0.22-5           
 [81] bit_4.0.5                 tidyselect_1.2.0         
 [83] GO.db_3.18.0              ComplexHeatmap_1.18.1    
 [85] locfit_1.5-9.8            Biostrings_2.70.2        
 [87] statmod_1.5.0             stringi_1.8.3            
 [89] lazyeval_0.2.2            evaluate_0.23            
 [91] codetools_0.2-19          tibble_3.2.1             
 [93] BiocManager_1.30.22       cli_3.6.2                
 [95] org.Ce.eg.db_3.18.0       IRkernel_1.3.2           
 [97] repr_1.1.6                munsell_0.5.0            
 [99] Rcpp_1.0.11               gprofiler2_0.2.2         
[101] png_0.1-8                 parallel_4.3.2           
[103] readr_2.1.5               blob_1.2.4               
[105] sparseMatrixStats_1.14.0  bitops_1.0-7             
[107] viridisLite_0.4.2         scales_1.3.0             
[109] purrr_1.0.2               crayon_1.5.2             
[111] GetoptLong_1.0.5          rlang_1.1.3              
[113] cowplot_1.1.3             KEGGREST_1.42.0        
```

Any clue would be appreciated!

**Comment by Ylefol** [comment]
Hello,

I'm a bit stumped by this one. I have updated my own environment to best resemble yours and things are still running with the PBMC example data. This leads me to believe that the problem is related with your data. Which would be good news as that will be a much easier solve than an issue with ComplexHeatmap!

Could you first try running the entire rmarkdown with the PBMC example data? If this works, we will know the source of the issue and we can then figure out how to solve it.

Best,
Yohan

**Comment by fpichon** [comment]
Hi @Ylefol,

Thanks for your answer and sorry for the delay.
So, I could find time to test the function with PBMC data as described at this link: https://ylefol.github.io/TimeSeriesAnalysis/articles/TS_analysis_PBMC.html

And I've got the same error:
`Error in grid.Call.graphics(C_downvppath, name$path, name$name, strict): Viewport 'log10(counts)_heatmap_body_1_1' was not found`

EDIT: I have same error with `DE_plots_vignettes(TS_object)`

**Comment by Ylefol** [comment]
Hello,

In that case it seems likely that another package may somehow be interfering, and thus causing an element to be missing/deleted, in this case 'log10(counts)[...]'. I'll look into this. 
To help me out, could you return to the example and save the TS_object before the first 'plot_wrapper_DE_results()' call, and send me the object (email is yohanl@medisin.uio.no).
This could help me pin point where the error is occurring and how to solve it.

Best,
Yohan

**Comment by fpichon** [comment]
OK, done!
Thanks!
Fabien

**Comment by Ylefol** [comment]
To update this issue. 
We were able to isolate that the issue was on a specific build as it was not reproducible on other builds/computers. 

Unfortunately the exact error was not solved and the solution was to remove/comment out the ComplexHeatmap component of the pipeline. Fortunately this only affects one visualization but not the end results of the pipeline.

Closing this issue for now, it may be re-opened if this error re-appears for another user or if additional information is found concerning this error.

Yohan
